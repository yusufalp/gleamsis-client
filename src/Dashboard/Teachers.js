import React from 'react';
import STORE from '../STORE/STORE_alt';

class Teachers extends React.Component {
  render() {
    let teachers = STORE.teachers.find(teacher => teacher.id === this.props.course.teacher_id)
    let course = this.props.course
    return (
      <tbody>
        <tr>
          <td>{teachers.firstName} </td>
          <td>{teachers.lastName}</td>
          <td>
            <p
              className="course-names">
              <a href={`/courses/${course.id}`}>
                {course.name}
              </a>
            </p>
          </td>
        </tr>
      </tbody>
    )
  }
}

export default Teachers;