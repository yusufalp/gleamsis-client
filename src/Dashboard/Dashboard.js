import React from 'react';
import { Link } from 'react-router-dom';
import DashMessage from '../Tables/MessageTable';
import ExpenditureTable from '../Tables/ExpenditureTable';
import Teachers from '../Teachers/Teachers';
import DataContext from '../DataContext';

class Dashboard extends React.Component {
  static contextType = DataContext
  render() {
    return (
      <div>
        <header>
          <h2>Welcome to your Dashboard</h2>
        </header>
        <div className='group'>
          <div className='item-dash'>
            <h3>Messages</h3>
            <DashMessage />
          </div>
          <div className='item-dash'>
            <h3>Expenditure</h3>
            <ExpenditureTable />
          </div>
        </div>
        <div className='group'>
          <div className='item-dash'>
            <h3>Courses</h3>
            <table>
              <thead>
                <tr>
                  <th>First Name</th>
                  <th>Last Name</th>
                  <th colSpan='3'>Courses</th>
                </tr>
              </thead>
              <tbody>
                {this.context.courses.map((course, i) =>
                  <Teachers key={i} course={course} />
                )}
              </tbody>
            </table>
            <Link to={'/add-teacher'} className='btn-add'>Add Teacher</Link>
            <Link to={'/add-course'} className='btn-add'>Add Course</Link>
          </div>
        </div>
      </div>
    )
  }
}

export default Dashboard;