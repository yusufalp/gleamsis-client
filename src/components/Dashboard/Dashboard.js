import React from 'react';
import { Link } from 'react-router-dom';
import Teachers from '../Teachers/Teachers';
import DataContext from '../DataContext';
import Inbox from '../Inbox/Inbox';
import Expenditures from '../Expenditures/Expenditures';

class Dashboard extends React.Component {
  static contextType = DataContext;
  render() {
    return (
      <div>
        <header>
          <h2>Welcome to your Dashboard</h2>
        </header>
        <div className='group'>
          <div className='item-dash card-display inbox-table'>
            <Inbox />
          </div>
          <div className='item-dash card-display exp-table'>
            <Expenditures />
          </div>
        </div>
        <div className='group'>
          <div className='item-dash card-display'>
            <h3>Courses</h3>
            <table>
              <thead>
                <tr>
                  <th>First Name</th>
                  <th>Last Name</th>
                  <th colSpan='3'>Courses</th>
                </tr>
              </thead>
              <tbody>
                {!this.context.courses.error ? this.context.courses.map((course, i) =>
                  <Teachers key={i} course={course} />
                ) : ""}
              </tbody>
            </table>
            <Link to={'/add-teacher'} className='btn-add'>Add Teacher</Link>
            <Link to={'/add-course'} className='btn-add'>Add Course</Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Dashboard;