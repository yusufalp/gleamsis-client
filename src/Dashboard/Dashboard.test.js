import React from 'react';
import ReactDOM from 'react-dom';
import Dashboard from './Dashboard';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('Dashboard component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Dashboard />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})