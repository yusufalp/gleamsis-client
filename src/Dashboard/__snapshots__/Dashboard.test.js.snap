// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders the UI as expected 1`] = `
<div>
  <header>
    <h2>
      Welcome to your Dashboard
    </h2>
  </header>
  <div
    className="group"
  >
    <div
      className="item-dash"
    >
      <h3>
        Messages
      </h3>
      <table>
        <thead>
          <tr>
            <th>
              From
            </th>
            <th>
              Subject
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Yusuf Alp
            </td>
            <td>
              Requesting purchase approval
            </td>
          </tr>
          <tr>
            <td>
              Thinkful
            </td>
            <td>
              Congratulations!
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="item-dash"
    >
      <h3>
        Expenditure
      </h3>
      <table>
        <thead>
          <tr>
            <th>
              Description
            </th>
            <th>
              Budget
            </th>
            <th>
              Actual
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Teacher Salary
            </td>
            <td>
              $84,000
            </td>
            <td>
              $87,432
            </td>
          </tr>
          <tr>
            <td>
              Books
            </td>
            <td>
              $23,135
            </td>
            <td>
              $23,246
            </td>
          </tr>
          <tr>
            <td>
              Food
            </td>
            <td>
              $3453
            </td>
            <td>
              $3453
            </td>
          </tr>
          <tr>
            <td>
              Sanitation
            </td>
            <td>
              $2500
            </td>
            <td>
              $4500
            </td>
          </tr>
        </tbody>
        <thead>
          <tr>
            <th>
              TOTAL
            </th>
            <th>
              $113,088
            </th>
            <th>
              $118,631
            </th>
          </tr>
        </thead>
      </table>
    </div>
  </div>
  <div
    className="group"
  >
    <div
      className="item-dash"
    >
      <h3>
        Courses
      </h3>
      <table>
        <thead>
          <tr>
            <th>
              First Name
            </th>
            <th>
              Last Name
            </th>
            <th
              colSpan="3"
            >
              Courses
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
      <button
        type="submit"
      >
        <a
          href="/add-teacher"
          onClick={[Function]}
        >
          Add Teacher
        </a>
      </button>
      <button
        type="submit"
      >
        <a
          href="/add-course"
          onClick={[Function]}
        >
          Add Course
        </a>
      </button>
    </div>
  </div>
</div>
`;
